### PR TITLE
PD-42 Implement Undo, Redo Commands

### PR DESCRIPTION
**Pull request by crapper**

### CODE REVIEW

File: src/core/MainApp.ts [hunk 1]
```diff
@@ -1,5 +1,61 @@
...
+  /**
+   * a method that eliminates all history and set the flag `isModified` to false
+   */
+  resetHistory() {
+    super.resetHistory();
+    this.sourceCurrentMemento = this.getLatestMemento();
+    this.isModifiedFlag = false;
+  }
+
+  // diagram: Diagram = new Diagram();
```
Comment (Jerrylum) [inline src/core/MainApp.ts:59]: Remove unused code

**Review by Jerrylum — CHANGES_REQUESTED** [review]

File: src/core/MainApp.test.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Diagram } from "../diagram/Diagram";
+import { MainApp } from "./MainApp";
+// import { MainDiagramHandler } from "./MainDiagramHandler";
```
Comment (Jerrylum) [inline src/core/MainApp.test.ts:3]: Remove unused code

**Review by Jerrylum — CHANGES_REQUESTED** [review]

File: src/core/MainApp.test.ts (hunk 2 above)
Comment (crapper) [reply src/core/MainApp.test.ts:3]: removed

**Review by Jerrylum — APPROVED** [review]
LGTM

**Pull request merged**